pkg/kubectl/cmd/get/get_test.go

```diff
@@ -1301,18 +1303,205 @@ func TestWatchResource(t *testing.T) {
-foo    0/0              0          <unknown>
-foo    0/0              0          <unknown>
+	expected := `EVENT      NAME   READY   STATUS   RESTARTS   AGE
+ADDED      foo    0/0              0          <unknown>
+MODIFIED   foo    0/0              0          <unknown>
+DELETED    foo    0/0              0          <unknown>
 `
 	if e, a := expected, buf.String(); e != a {
 		t.Errorf("expected\n%v\ngot\n%v", e, a)
 	}
 }
 
+func TestWatchResourceAllNamespaces(t *testing.T) {
+
+	testcases := []struct {
+		format   string
+		expected string
+	}{
+		{
+			format: "",
+			expected: `EVENT      NAMESPACE   NAME      READY   STATUS   RESTARTS   AGE
+ADDED      test        pod/bar   0/0              0          <unknown>
+ADDED      test        pod/foo   0/0              0          <unknown>
+MODIFIED   test        pod/foo   0/0              0          <unknown>
+DELETED    test        pod/foo   0/0              0          <unknown>
+`,
+		},
+		{
+			format: "wide",
+			expected: `EVENT      NAMESPACE   NAME      READY   STATUS   RESTARTS   AGE         IP       NODE     NOMINATED NODE   READINESS GATES
+ADDED      test        pod/bar   0/0              0          <unknown>   <none>   <none>   <none>           <none>
+ADDED      test        pod/foo   0/0              0          <unknown>   <none>   <none>   <none>           <none>
+MODIFIED   test        pod/foo   0/0              0          <unknown>   <none>   <none>   <none>           <none>
+DELETED    test        pod/foo   0/0              0          <unknown>   <none>   <none>   <none>           <none>
+`,
+		},
+		{
+			format: "json",
+			expected: `{"type":"ADDED","object":{"apiVersion":"v1","kind":"Pod","metadata":{"creationTimestamp":null,"name":"bar","namespace":"test","resourceVersion":"9"},"spec":{"containers":null,"dnsPolicy":"ClusterFirst","enableServiceLinks":true,"restartPolicy":"Always","securityContext":{},"terminationGracePeriodSeconds":30},"status":{}}}
+{"type":"ADDED","object":{"apiVersion":"v1","kind":"Pod","metadata":{"creationTimestamp":null,"name":"foo","namespace":"test","resourceVersion":"10"},"spec":{"containers":null,"dnsPolicy":"ClusterFirst","enableServiceLinks":true,"restartPolicy":"Always","securityContext":{},"terminationGracePeriodSeconds":30},"status":{}}}
+{"type":"MODIFIED","object":{"apiVersion":"v1","kind":"Pod","metadata":{"creationTimestamp":null,"name":"foo","namespace":"test","resourceVersion":"11"},"spec":{"containers":null,"dnsPolicy":"ClusterFirst","enableServiceLinks":true,"restartPolicy":"Always","securityContext":{},"terminationGracePeriodSeconds":30},"status":{}}}
+{"type":"DELETED","object":{"apiVersion":"v1","kind":"Pod","metadata":{"creationTimestamp":null,"name":"foo","namespace":"test","resourceVersion":"12"},"spec":{"containers":null,"dnsPolicy":"ClusterFirst","enableServiceLinks":true,"restartPolicy":"Always","securityContext":{},"terminationGracePeriodSeconds":30},"status":{}}}
+`,
+		},
+		{
+			format: "yaml",
```

Comment:
not sure what to do here... the printer for a single object doesn't know there are more objects coming, so printing a separator doesn't seem quite right

Reply:
yeah, I think we definitely need a separator.  PrintSingleObject is the bane of our existence, but it should be possible to wrap printSingleObject with a stateful printer that knows when you previously printed (no separator for first object, add separator for all subsequent).